server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateDataStreamService.java

```diff
@@ -158,4 +165,16 @@ public static ComposableIndexTemplate lookupTemplateForDataStream(String dataStr
         return composableIndexTemplate;
     }
 
+    public static void validateTimestampFieldMapping(String timestampFieldName, MapperService mapperService) {
+        MappedFieldType timestampFieldMapper = mapperService.fieldType(timestampFieldName);
+        if (timestampFieldMapper == null) {
+            throw new IllegalArgumentException("expected timestamp field [" + timestampFieldName + "], but found no timestamp field");
+        }
+        String type = timestampFieldMapper.typeName();
+        if (ALLOWED_TIMESTAMPFIELD_TYPES.contains(type) == false) {
+            throw new IllegalArgumentException("expected timestamp field [" + timestampFieldName + "] to be of types [" +
+                ALLOWED_TIMESTAMPFIELD_TYPES + "], but instead found type [" + type  + "]");
```

Comment:
Super minor:

```suggestion
            throw new IllegalArgumentException("expected timestamp field [" + timestampFieldName + "] to be of types " +
                ALLOWED_TIMESTAMPFIELD_TYPES + ", but instead found type [" + type  + "]");
```

To avoid the `[[]]` double brackets